app/src/main/java/com/example/habithelper/activities/AccountCreationActivity.java

```diff
@@ -39,113 +37,138 @@ public class SetupAccount extends AppCompatActivity {
-                }
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JSON json) {
+                validZip = true;
+                tryToInitialize(validZip);
+            }
 
-                if (habitsList.size()>10) {
-                    Toast.makeText(SetupAccount.this, "Select fewer habits", Toast.LENGTH_SHORT).show();
-                    habitsList.clear();
-                    return;
-                }
+            @Override
+            public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+                validZip = false;
+                tryToInitialize(validZip);
+                return;
+            }
+        });
+    }
 
-                validZip=false;
-                // checking whether or not the zipcode inputted is valid
-                AsyncHttpClient client = new AsyncHttpClient();
-                String api_request=GET_WEATHER_URL+zipString;
-                client.get(api_request, new JsonHttpResponseHandler() {
-
-                    @Override
-                    public void onSuccess(int statusCode, Headers headers, JSON json) {
-                        validZip=true;
-                        tryToInitialize(validZip);
-                    }
+    /**
+     * Checks whether the name, custom habits, zipcode, and number of habits the user selected are valid
+     */
+    private void checkUserInput() {
+        name = etName.getText().toString();
+        zipString = etZip.getText().toString();
+        customHabitOne = etCustomOne.getText().toString();
+        customHabitTwo = etCustomTwo.getText().toString();
+
+        if (checkBoxCustomOne.isChecked()) {
+            if (customHabitOne.isEmpty()) {
+                Toast.makeText(AccountCreationActivity.this, "habit cannot be empty", Toast.LENGTH_SHORT).show();
+                return;
+            }
+        }
 
-                    @Override
-                    public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
-                        validZip=false;
-                        tryToInitialize(validZip);
-                        return;
-                    }
-                });
+        if (checkBoxCustomTwo.isChecked()) {
+            if (customHabitTwo.isEmpty()) {
+                Toast.makeText(AccountCreationActivity.this, "habit cannot be empty", Toast.LENGTH_SHORT).show();
+                return;
+            }
+        }
 
+        if (name.isEmpty()) {
+            Toast.makeText(AccountCreationActivity.this, "name cannot be empty", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (zipString.isEmpty()) {
+            Toast.makeText(AccountCreationActivity.this, "zipcode cannot be empty", Toast.LENGTH_SHORT).show();
+            return;
+        }
 
+        // initializing the list of habits the user wishes to track
+        initializeHabitList();
 
+        if (habitsList.size() < 4) {
```

Comment:
Another good practice is to use a list of constants for this hardcoded values and you can have all these constants at the top of the file, or on a different file if they are being shared across other places, like:
MIN_REQUIRED_HABITS_NUM = 4